packages/js/product-editor/src/components/attribute-term-input-field/attribute-term-input-field.tsx

```diff
@@ -133,6 +120,55 @@ export const AttributeTermInputField: React.FC<
 		}
 	};
 
+	const createAttributeTerm = async (
+		attribute: Partial< ProductAttributeTerm >
+	) => {
+		recordEvent( 'product_attribute_term_add', {
+			source: TRACKS_SOURCE,
+		} );
+		setIsCreatingTerm( true );
+		try {
+			const newAttribute: ProductAttributeTerm =
+				await createProductAttributeTerm( {
+					...attribute,
+					attribute_id: attributeId,
+				} );
+			recordEvent( 'product_attribute_term_add_success', {
+				source: TRACKS_SOURCE,
+			} );
+			onChange( [ ...value, newAttribute ] );
+			invalidateResolutionForStoreSelector( 'getProductAttributes' );
+			setIsCreatingTerm( false );
+		} catch ( e ) {
+			recordEvent( 'product_attribute_term_add_failed', {
+				source: TRACKS_SOURCE,
```

Comment:
We should add the error code to this and the error message, if it exists.

Reply:
Good call! I added it in the commit [93fedd6](https://github.com/woocommerce/woocommerce/pull/39610/commits/93fedd6709933d59661696727f29028147052ece)